File: SpotifyAlarmClock/Controllers/StartScreen/StartScreenViewController.m
//
//  StartScreenViewController.m
//  Alarm Clock
//
//  Created by Niels Vroegindeweij on 23-11-14.
//  Copyright (c) 2014 Niels Vroegindeweij. All rights reserved.
//

#import "StartScreenViewController.h"

@interface StartScreenViewController ()

- (IBAction)skipLoginButtonClicked:(id)sender;

@end

@implementation StartScreenViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setNeedsStatusBarAppearanceUpdate];
    
    UIImageView *tempImageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"LoginBackground"]];
    [tempImageView setFrame:self.tableView.frame];
   
    self.tableView.backgroundView = tempImageView;
}

-(UIStatusBarStyle)preferredStatusBarStyle{
    return UIStatusBarStyleLightContent;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewWillAppear:(BOOL)animated {
    [self.navigationController setNavigationBarHidden:YES animated:animated];
    [super viewWillAppear:animated];
    
    [[UIDevice currentDevice] setValue:
     [NSNumber numberWithInteger: UIInterfaceOrientationPortrait]
                                forKey:@"orientation"];
}

- (void)viewWillDisappear:(BOOL)animated {
    [self.navigationController setNavigationBarHidden:NO animated:animated];
    [super viewWillDisappear:animated];
}



- (IBAction)skipLoginButtonClicked:(id)sender
{
    UIAlertView *alert = [[UIAlertView alloc]initWithTitle:@"Spotify Usage"
                                                   message:@"Are you sure you want to use the Alarm Clock without Spotify features? You can enter your credentials afterwards in the settings menu."
                                                  delegate:self
                                         cancelButtonTitle:@"No"
                                         otherButtonTitles:@"Yes",nil];
    [alert show];
}

- (void) alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex
{
    if (buttonIndex == [alertView firstOtherButtonIndex])
    {
        [[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"UseAlarmClockWithoutSpotify"];
        [self dismissViewControllerAnimated:YES completion:nil];
    }
}

#pragma mark - Table view delegate
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    switch ([indexPath row]) {
        case 1: //Linked In
            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"https://www.spotify.com/nl/signup/"]];
    }
}
- (void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath
{
    [cell setBackgroundColor:[UIColor clearColor]];
}
@end

File: SpotifyAlarmClock/Controllers/StartScreenViewController.swift
//
//  StartScreenViewController.swift
//  Alarm Clock
//
//  Created by Niels Vroegindeweij on 10-03-15.
//  Copyright (c) 2015 Niels Vroegindeweij. All rights reserved.
//

import UIKit

class StartScreenViewController: UITableViewController, UIAlertViewDelegate {

    @IBOutlet weak var loginCell : UITableViewCell!
    @IBOutlet weak var noSpotifyCell : UITableViewCell!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.setNeedsStatusBarAppearanceUpdate()
        
        //Set background
        let tempImageView = UIImageView(image: UIImage(named: "LoginBackground"))
        tempImageView.frame = self.tableView.frame
        self.tableView.backgroundView = tempImageView;
        
        //Set disclosure indicators
        loginCell.accessoryView = UIImageView(image: UIImage(named: "DisclosureIndicator"))
        noSpotifyCell.accessoryView = UIImageView(image: UIImage(named: "DisclosureIndicator"))
        
        //Set select cell background
        let bgColorView = UIView()
        bgColorView.backgroundColor = UIColor.grayColor()
        loginCell.selectedBackgroundView = bgColorView
        noSpotifyCell.selectedBackgroundView = bgColorView
    }
    
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return UIStatusBarStyle.LightContent
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        self.navigationController?.setNavigationBarHidden(true, animated: animated)
        UIDevice.currentDevice().setValue(UIInterfaceOrientation.Portrait.rawValue, forKey: "orientation")
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        
        self.navigationController?.setNavigationBarHidden(false, animated: animated)
    }

    // MARK: - Alert view delegate
    func alertView(alertView: UIAlertView, clickedButtonAtIndex buttonIndex: Int) {
        if buttonIndex == 1 {
            NSUserDefaults.standardUserDefaults().setBool(true, forKey: "UseAlarmClockWithoutSpotify")
            self.dismissViewControllerAnimated(true, completion: nil)
        }
    }
    
    // MARK: - Table view delegate
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
        
        switch indexPath.row {
        case 1: //Linked In
            let alert = UIAlertView(title: "Spotify Usage",
                                    message: "Are you sure you want to use the Alarm Clock without Spotify features? You can set your credentials afterwards via the settings menu.",
                                    delegate: self,
                                    cancelButtonTitle: "No",
                                    otherButtonTitles: "Yes")

            alert.show()
        default:
            break
        }
    }
    
    override func tableView(tableView: UITableView, willDisplayCell cell: UITableViewCell, forRowAtIndexPath indexPath: NSIndexPath) {
        cell.backgroundColor = UIColor.clearColor()
    }

}
